### PR TITLE
Fix validation errors on lists page

### LINKED ISSUE #4472: Lists page has duplicate lists id
When using the validator tool on https://validator.w3.org/nu/?doc=https%3A%2F%2Fopenlibrary.org%2Flists you'll notice that one of the complaints is that #list is used twice.

* Ideally remove the id on the body tag, or at least rename it to "page-lists"
* Confirm the validator passes for this page.

### Evidence / Screenshot (if possible)

Screen Shot 2021-01-23 at 9 32 08 AM

### Relevant url?
https://openlibrary.org/lists

### Stakeholders
@jdlrobson

### PR DESCRIPTION
**Pull request by Yashs911**
Closes #4472

### Stakeholders

@jdlrobson

### CODE REVIEW

File: bundlesize.config.json [hunk 1]
```diff
@@ -34,11 +34,11 @@
     },
     {
       "path": "static/build/page-edit.css",
-      "maxSize": "20KB"
+      "maxSize": "21KB"
```
Comment (jdlrobson) [inline bundlesize.config.json:37]: These are not related to this patchset, please rebase now that the master branch has been cleaned up and remove this change.

File: openlibrary/templates/lists/home.html [hunk 2]
```diff
@@ -26,7 +26,7 @@ <h2>$_('Create a list of any Subjects, Authors, Works or specific Editions.')</h
     <div class="head">
         <h2>$_('Active Lists')</h2>
     </div>
-    <div id="lists">
+    <div id="user-lists">
```
Comment (jdlrobson) [inline openlibrary/templates/lists/home.html:29]: Why is an ID needed for this at all? Since #lists is only used for styling purposes, it would be better if making this change to resort to using a class name.

**Review by jdlrobson — CHANGES_REQUESTED** [review]
The ID on the body tag is also problematic IMO, and we should seek to remove it as part of these changes.

File: openlibrary/templates/lists/home.html (hunk 2 above)
Comment (Yashs911) [reply openlibrary/templates/lists/home.html:29]: Okay but the ID on the body tag is actually generated and is interconnected in some way. Changing the id breaks the page.

```
class lists_home(delegate.page):
    path = "/lists"

    def GET(self):
        delegate.context.setdefault('bodyid', 'lists')
        return render_template("lists/home")
```
So I don't think we can remove it

Comment (jdlrobson) [reply openlibrary/templates/lists/home.html:29]: https://github.com/internetarchive/openlibrary/blob/07e7346293027209f0e75961e7b3d0b0b6e86e1e/openlibrary/templates/site/body.html#L8

The body id is added as an attribute and is no different from a template. We should omit it there. There's no need for a body to have an id - there's only ever gonna be one and it can be accessed via document.body

File: bundlesize.config.json (hunk 1 above)
Comment (Yashs911) [reply bundlesize.config.json]: Done.

**Comment by Yashs911** [comment]
@jdlrobson, I have made the required changes.

File: openlibrary/templates/site/body.html [hunk 3]
```diff
@@ -5,7 +5,7 @@
 $ show_banners = bodyid != 'form'
 $ bodyattrs = ctx.get('bodyattrs', [])
 
-<body id="$bodyid" class="$(' '.join(bodyclass))" $:(' '.join(bodyattrs))>
+<body class="$(' '.join(bodyclass))" $:(' '.join(bodyattrs))>
```
Comment (jdlrobson) [inline openlibrary/templates/site/body.html:8]: Not this easy I'm afraid :)

Run `make css` and you'll see some CSS regressions on quite a few pages.

There are various CSS selectors e.g. `body#plain` that need to be fixed.

The good news is you can remove most of these, e.g. `body#plain div#headerLogo` becomes `div#headerLogo` but be careful about any generic selectors. `body#plain div` should not become `div`. 

Note we have page stylesheet entry points so a rule `body#plain div` can be moved into static/css/page-plain.less` if necessary.

**Review by jdlrobson — CHANGES_REQUESTED** [review]
Please update body# selectors in CSS .

**Comment by Yashs911** [comment]
@jdlrobson, some styles are directly applied to the body# selectors, so how should I go about it?
image

**Comment by jdlrobson** [comment]
> @jdlrobson, some styles are directly applied to the body# selectors, so how should I go about it?

Note the body id selectors predate the page specific CSS. `page-form` is currently only ever applying to pages with the ID form on the body, so this means you can safely do

```
body {
 text-align: left;
 ...
}
```
For the nested CSS, it should be fine to move them out of the body tag and remove a level of indentation.

**Comment by Yashs911** [comment]
@jdlrobson I have moved the internal CSS. The color mentioned are not present in colors.less so should I add them?

I was wondering how all these CSS are linked to the template.html files?

**Comment by jdlrobson** [comment]
> @jdlrobson I have moved the internal CSS. The color mentioned are not present in colors.less so should I add them?
> 
> I was wondering how all these CSS are linked to the template.html files?

The magic happens in openlibrary/templates/site/head.html
The bodyid is used to generate the name of the stylesheet to be added to the page.

Ideally we'd have less of these page specific stylesheets but there's a lot of clean up to do first.

**Review by jdlrobson — CHANGES_REQUESTED** [review]
There are some linting issues. Run `npm run lint` locally.

Will pull this down for testing once that's done.

**Comment by Yashs911** [comment]
@jdlrobson what about color-no-hex rule?

**Comment by jdlrobson** [comment]
> @jdlrobson what about color-no-hex rule?

Make sure you use one of the variables in static/css/less/colors.less. Add a new one if necessary.

**Comment by Yashs911** [comment]
@jdlrobson Now you can review this PR

File: static/css/legacy-borrowTable-adminUser.less [hunk 4]
```diff
@@ -5,30 +5,77 @@
...
       }
     }
   }
-  /* stylelint-enable selector-max-specificity */
+}
+tr {
+  &.row-available {
+    background: @admin-row-background !important;
+    .status-value {
+      color: @dark-green-two;
```
Comment (jdlrobson) [inline static/css/legacy-borrowTable-adminUser.less:35]: Let's use @dark-green here rather than create a new variable. The existing variable already has a confusing name :)

We can always change those colors later.

File: openlibrary/templates/borrow_admin.html [hunk 5]
```diff
@@ -341,75 +335,3 @@ <h4>Waiting List JSON</h4>
...
-    font-family:"News Gothic MT","Trebuchet MS",Geneva,Helvetica,sans-serif;
-    color:#615132;
-    font-weight:600;
-    font-size:0.6875em;
-    text-transform: uppercase;
-    white-space: nowrap;
-    text-align: center;
-}
-
-th.xtitles {
```
Comment (jdlrobson) [inline openlibrary/templates/borrow_admin.html:362]: This rule appears to be lost. Was this intentional?

File: static/css/legacy-borrowTable-adminUser.less [hunk 6]
```diff
@@ -5,30 +5,77 @@
...
+      white-space: nowrap;
+      text-align: center;
+      &.titles {
+        text-align: left;
       }
     }
   }
-  /* stylelint-enable selector-max-specificity */
+}
+tr {
```
Comment (jdlrobson) [inline static/css/legacy-borrowTable-adminUser.less:31]: All these rules beginning here should be nested under #borrowTable. Otherwise they apply globally.

**Review by jdlrobson — CHANGES_REQUESTED** [review]

File: openlibrary/templates/borrow_admin.html (hunk 5 above)
Comment (Yashs911) [reply openlibrary/templates/borrow_admin.html:362]: &.titles {
        text-align: left;
      }
I thought that "x" was typo.

Comment (Yashs911) [reply openlibrary/templates/borrow_admin.html:362]: https://github.com/internetarchive/openlibrary/pull/4482#diff-ebde383b40e87da92da077076dab958e61e8bedb93d959b16d5d1501234dfb8dR25

Comment (jdlrobson) [reply openlibrary/templates/borrow_admin.html:362]: appears to be used on L102 and L166

File: static/css/legacy-borrowTable-adminUser.less (hunk 6 above)
Comment (Yashs911) [reply static/css/legacy-borrowTable-adminUser.less]: Done.

File: static/css/page-plain.less [hunk 7]
```diff
@@ -12,187 +12,185 @@
...
+    border-bottom: none;
   }
-  // Specific to borrowTable component
-  /* stylelint-disable selector-max-specificity */
-  #borrowTable {
-    margin: 60px 0 25px;
+}
+// Specific to borrowTable component
+/* stylelint-disable selector-max-specificity */
+#borrowTable {
```
Comment (jdlrobson) [inline static/css/page-plain.less:148]: Perhaps in a follow up we could move this into legacy-borrowTable-plain.less for consistency with `legacy-borrowTable-adminUser.less`.

The eventual goal would be to unify these styles with those in legacy-borrowTable-adminUser.less  and static/css/components/borrowTable.less

**Review by jdlrobson — APPROVED** [review]
